### ISSUE TITLE
Debugging credential issue?

### ISSUE THREAD

**Issue by mbenua-mparticle** [wepay/TeamCity-Phabricator-Plugin#16]
Hi guys, do you have any log output (or way to view log output)? I'm trying to debug a maddening credential issue using your plugin. I can make HTTP posts to /httpAuth/app/rest/buildQueue on my TeamCity install using basic HTTP auth just fine when I use Fiddler, but when I give Phabricator those same credentials TeamCity rejects the build request with incorrect username/password. Without having any insight into what Phabricator is actually doing, I'm not sure where to go from here - any ideas you guys have for debugging would be much appreciated! A quick inspection of the loaded Phabricator php plugin looks fine, and I've triple-checked the username/password in Phabricator to make sure it matches up with what I'm using manually.

I'm suspicious something in here isn't right (it's straight from your master), but not familiar enough with Phabricator PHP development to go much further.
 ```
   $credential_phid = $this->getSetting('credential');
    if ($credential_phid) {
       $key = PassphrasePasswordKey::loadFromPHID(
              $credential_phid,
              $viewer);
       $username = $key->getUsernameEnvelope()->openEnvelope();
       $password = $key->getPasswordEnvelope();
       curl_setopt($process, CURLOPT_USERPWD, $username . ":" . $password);
    }
```

**Comment by mbenua-mparticle** [comment]
Never mind, there are some issues with your PHP file. I've rolled back to using the original. I'm happy to share the final output if you guys are interested.

**Comment by dclarke** [comment]
Mbenua would be glad to see your results / why you rolled back to the original. 
In our scenario we ran into an incompatibility because we had an older version of phabricator that did not have the build_target functionality. 
I think this should not have broken your build though.. we have since upgraded phabricator with no problems found.

**Comment by dclarke** [comment]
What version of phabricator are you running ? 
Does the curl request fail ?

**Comment by mbenua-mparticle** [comment]
We're using the most recent version of Phabricator (it's a pretty fresh
install) and 10.x of TeamCity. There are two issues:
1) yes, the curl request is failing - why did you move away from using the
inbuilt Phabricator HTTP wrapper? That seems to work perfectly.
2) you had some log lines commented out, but hadn't commented out line -
list($headers, $body) = explode("\r\n\r\n", $return, 2); - for whatever
reason, php was throwing an exception on this line that only became
apparent when I ran harbormaster manually from the command

I actually suspect 2) is also due to curl usage, because things like status
code were also returning as 0 after I removed the explode line - so $return
may be null.

To be clear, when I moved to the original master from the halfmatthalfcat
repo, the communication works perfectly between the two servers.
